improver/calibration/rainforest_calibration.py

```diff
@@ -404,40 +402,56 @@ def _evaluate_probabilities(
         self,
         forecast_data: ndarray,
         input_data: ndarray,
+        forecast_variable: str,
+        forecast_variable_unit: str,
         output_data: ndarray,
-        preprocessing_fun: Optional[Callable[[ndarray], object]] = None,
+        model_input_converter: Optional[Callable[[ndarray], object]] = None,
     ):
         """Evaluate probability that error in forecast exceeds thresholds, setting
-        the result to 1 when `forecast + threshold <= 0`.
+        the result to 1 when `forecast + threshold` is less than or equal to
+        the lower bound of forecast_variable, as defined in constants.BOUNDS_FOR_ECDF`.
 
         Args:
             forecast_data:
                 1-d containing data for the variable to be calibrated.
             input_data:
                 2-d array of data for the feature variables of the model
+            forecast_variable:
+                name of forecast variable
+            forecast_variable_unit:
+                unit of forecast variable
             output_data:
                 array to populate with output; will be modified in place
-            preprocessing_fun:
+            model_input_converter:
                 function to convert data from ndarray to model input format,
                 only used for treelite predictor
         """
 
-        if preprocessing_fun:
-            input_dataset = preprocessing_fun(input_data)
+        if model_input_converter:
+            input_dataset = model_input_converter(input_data)
         else:
             input_dataset = input_data
 
+        bounds_data = BOUNDS_FOR_ECDF[forecast_variable]
+        bound_unit = unit.Unit(bounds_data[1])
+        lower_bound = bound_unit.convert(bounds_data[0][0], forecast_variable_unit)
+
         for threshold_index, model in enumerate(self.tree_models):
             threshold = self.error_thresholds[threshold_index]
             if threshold >= 0:
```

Comment:
Does this inequality need to change to incorporate the use of `lower_bound`?

Reply:
No; the logic is:
```
error > threhsold
iff obs - fcst > threshold
iff obs > threshold + fcst
```
If `threshold >= 0` then, provided `fcst >= lower_bound` (which should always be true), we also have `threshold + fcst >= lower_bound`, i.e. `threshold + fcst` is always a plausible value of the variable and it's possible for the observation to be above or below it, so we need to run the prediction model for each row.
However if `threshold < 0`, then `threshold + fcst` can be less than `lower_bound`, in which case it will definitely be less than the observed value and we do not need to run the prediction model.